### DISCUSSION TITLE
Hi addType() not available

### DISCUSSION THREAD

**Discussion by sunnystone85** [ruflin/Elastica#2203] [General]
Good day,

we are using Elastica 6.x before, now we upgrade to 7.x.
But it report an error as : call to undefined method Elastica\search::addType().
So the function addType() is not existed? is there any other function instead?

the code as below: 
		$search = new Elastica\Search($this->getClient());
		$search->addIndex($index_name)->addType($doc_type);

Thanks

**Comment by ruflin** [comment]
Are you aware of the type removal in Elasticsearch? https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html What version of Elasticsearch are you running?

**Reply by sunnystone85** [reply]
we are using ElasticSearch 5.2

**Comment by sunnystone85** [comment]
The Elasticsearch not upgraded, but due to our framework upgrade the dependency of Elastica need to upgrade to 7.x, but the function addType() was removed, just would like to find out a alternative way in Elastica 7.x...
Could you please help on this?

**Reply by ruflin** [reply]
Elastica 7.0 is not compatible with Elasticsearch 5.0: https://github.com/ruflin/Elastica What you can always do is send "raw" json requests to Elasticsearch with Elastica, like this you should be able to simulate any requests you need, but you are loosing all the Elastica Objects.

**Reply by sunnystone85** [reply]
Thanks for your reply...

**Reply by sunnystone85** [reply]
Hi Ruflin,
One more question, Is there any document/samples about sending "raw" json to Elasticsearch with Elastica?

**Reply by ruflin** [reply]
You can find here raw queries and raw JSON example: https://elastica.io/examples/ Be aware, these docs have not been touched for quite a while but it should still work.

**Reply by sunnystone85** [reply]
Great! Will have a try, thank you.
